clinica/pydra/t1_volume/t1_volume_utils.py

```diff
@@ -0,0 +1,536 @@
+
+    References
+    ------
+    https://brainder.org/2012/09/23/the-nifti-file-format/
+
+    """
+
+    from os.path import isfile
+
+    import nibabel as nib
+    import numpy as np
+    from nibabel.filebasedimages import ImageFileError
+
+    # from clinica.utils.stream import cprint
+    # assert isinstance(nii_volume, str), "input argument nii_volume must be a str"
+
+    assert isfile(nii_volume), "input argument must be a path to a file"
+
+    try:
+        orig_nifti = nib.load(str(nii_volume))
+    except ImageFileError:
+        print(f"File {nii_volume} could not be read by nibabel. Is it a valid NIfTI")
+        return np.nan
+
+    head = orig_nifti.header
+
+    if isinstance(head, nib.freesurfer.mghformat.MGHHeader):
+        # If MGH volume
+        center_coordinates_world = vox_to_world_space_method_3_bis(
+            head["dims"][0:3] / 2, head
+        )
+    else:
+        # Standard NIfTI volume
+        center_coordinates = get_center_volume(head)
+
+        if head["qform_code"] > 0:
+            center_coordinates_world = vox_to_world_space_method_2(
+                center_coordinates, head
+            )
+        elif head["sform_code"] > 0:
+            center_coordinates_world = vox_to_world_space_method_3(
+                center_coordinates, head
+            )
+        elif head["sform_code"] == 0:
+            center_coordinates_world = vox_to_world_space_method_1(
+                center_coordinates, head
+            )
+        else:
+            center_coordinates_world = np.nan
+    return center_coordinates_world
+
+
+def get_center_volume(header: dict) -> ndarray:
+    """Get the voxel coordinates of the center of the data, using header information.
+
+    Parameters
+    ----------
+        header: Nifti1Header
+            Contains image metadata
+
+    Returns
+    -------
+    ndarray
+        Voxel coordinates of the center of the volume
+    """
+    import numpy as np
+
+    center_x = header["dim"][1] / 2
+    center_y = header["dim"][2] / 2
+    center_z = header["dim"][3] / 2
+    return np.array([center_x, center_y, center_z])
+
+
+def vox_to_world_space_method_1(
+    coordinates_vol: ndarray, header: Nifti1Header
+) -> ndarray:
+    """Convert coordinates to world space
+
+    Parameters
+    ----------
+    coordinates_vol: ndarray
+        Coordinate in the volume (raw data)
+    header: Nifti1Header
+        Contains image metadata
+
+    Returns
+    -------
+    ndarray
+        Coordinates in the world space
+
+    Notes
+    -----
+    This method is for compatibility with analyze and is not supposed to be used as the main orientation method. But it
+    is used if sform_code = 0. The world coordinates are determined simply by scaling by the voxel size by their
+    dimension stored in pixdim.
+
+    References
+    ----------
+    https://brainder.org/2012/09/23/the-nifti-file-format/
+    """
+    import numpy as np
+
+    return np.array(coordinates_vol) * np.array(
```

Comment:
Can be simplified to :

```python
return np.array(coordinate_vol) * np.array([_ for _ in header["pixdim"][1:4]])
```

Reply:
I'd prefer to leave as is for it being more explicit. If you have a strong feeling about it, we can change.

Reply:
Be careful with `_`, this really means the variable is unpacked but **unused** later on. This is not the case here.